packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1426,6 +1450,13 @@ export class Project extends ProjectBase {
       return undefined;
     }
 
+    // !! Failsafe !! This should not occur with the current methods, given:
+    // * The fleet can only be set with the constructor
+    // * The batch builds can only be enabled with the `enableBatchBuilds` method
+    if (this.isBatchBuildEnabled) {
+      throw new Error('Build batch is not supported for project using reserved capacity');
+    }
```

Comment:
As mentioned in the comment, this condition should never be true. I've added it just in case a future CDK API change enables this behavior

Reply:
Agree, this is a good regression check.